rftp/file_response.go

```diff
@@ -45,11 +49,27 @@ func newFileResponse(index uint16) *FileResponse {
 		pwriter:       w,
 		buffer:        newChunkQueue(index),
 		resendEntries: make(map[uint64]struct{}),
+		hasher:        md5.New(),
 	}
 }
 
 func (f *FileResponse) Read(p []byte) (n int, err error) {
-	return f.preader.Read(p)
+	n, err = f.preader.Read(p)
+	if err != nil {
+		return
+	}
+	_, err = f.hasher.Write(p[:n])
+	if err != nil {
+		return
+	}
+	return
+}
+
+// Returns true iff the checksum of the received file matches the checksum
+// transmitted in the metadata message. Returns false while the file is still in
+// transit.
+func (f *FileResponse) ChecksumValid() bool {
+	return bytes.Compare(f.checksum[:], f.hasher.Sum(nil)[:16]) == 0
```

Comment:
The [`bytes.Compare`](https://golang.org/pkg/bytes/#Compare) documentation says *Prefer Equal to Compare for equality comparisons.* Is there a reason why you use `Compare` anyway?